### PR TITLE
Clean up types and providers

### PR DESCRIPTION
**Pull request by ekohl**
Closes GH-205
Closes GH-245

### CODE REVIEW

**Comment by ekohl** [comment]
One of the issues is now that the ```pulp_puppetrepo``` adds a plugin which means the DB needs to be updated. That requires all services to be stopped which we currently don't handle.

**Comment by ekohl** [comment]
Includes https://github.com/Katello/puppet-pulp/pull/248

File: lib/puppet/provider/pulp_rpmrepo/api.rb [hunk 1]
```diff
@@ -1,10 +1,18 @@
-require File.expand_path('../../../util/pulp_util', __FILE__)
+require File.expand_path('../../../util/repo_provider', __FILE__)
 
-Puppet::Type.type(:pulp_rpmrepo).provide(:api) do
+Puppet::Type.type(:pulp_rpmrepo).provide(:api, :parent => PuppetX::Pulp::RepoProvider) do
   commands :pulp_admin => '/usr/bin/pulp-admin'
 
   mk_resource_methods
 
+  def self.repo_type
```
Comment (ekohl) [inline lib/puppet/provider/pulp_rpmrepo/api.rb]: I think that this is very ugly, but I don't know enough about ruby inheritance to do this cleaner.

Comment (ekohl) [inline lib/puppet/provider/pulp_rpmrepo/api.rb]: (you can see my attempt at the last commit)

File: examples/pulp_isorepo.pp [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class { '::pulp::repo::upstream': } ->
+class { '::pulp':
+  # https://github.com/Katello/puppet-pulp/issues/138
+  ssl_username => '',
+  enable_admin => true,
+}
+
+pulp_isorepo { 'DataMonster':
+  display_name    => 'Data Monster',
+  description     => "Cookie's cousin",
```
Comment (ehelms) [inline examples/pulp_isorepo.pp]: I chuckled good.

**Comment by ehelms** [comment]
I don't see any major issues here, it still bothers me a bit this has to re-implement a Ruby API to talk to Pulp when we have Runcible. But if tests are passing I'm OK with all of this.

**Comment by ekohl** [comment]
Rebased on master, I think this is good for a final review and to be merged now.

**Comment by ekohl** [comment]
So I rebased this on master and the dependency on the unreleased qpid breaks acceptance tests. Does this module really need the unreleased version (and thus work around it by including a git checkout) or should we decrease the minimal version to one that's on the forge?

**Comment by ekohl** [comment]
https://github.com/Katello/puppet-pulp/pull/258 would unblock this.

**Comment by ekohl** [comment]
Looks like I screwed up some rebases and lost some work.

**Pull request merged**